accounts/abi/abi.go

```diff
@@ -132,13 +115,61 @@ func (abi ABI) Call(executer Executer, name string, args ...interface{}) interfa
 		return nil
 	}
 
-	output := executer(callData)
+	return abi.parseInterface(name, executer(callData))
+}
 
-	method := abi.Methods[name]
-	ret := make([]interface{}, int(math.Max(float64(len(method.Outputs)), float64(len(output)/32))))
-	for i := 0; i < len(ret); i += 32 {
-		index := i / 32
-		ret[index] = toGoType(method.Outputs[index], output[i:i+32])
+// toGoType parses the input and casts it to the proper type defined by the ABI
+// argument in T.
+func toGoType(i int, t Argument, output []byte) interface{} {
+	index := i * 32
+
+	// Parse the given index output and check whether we need to read
+	// a different offset and length based on the type (i.e. string, bytes)
+	var returnOutput []byte
+	switch t.Type.T {
+	case StringTy, BytesTy: // variable arrays are written at the end of the return bytes
+		// parse offset from which we should start reading
+		offset := int(common.BytesToBig(output[index : index+32]).Uint64())
+		// parse the size up until we should be reading
+		size := int(common.BytesToBig(output[offset : offset+32]).Uint64())
+		// get the bytes for this return value
+		returnOutput = output[offset+32 : offset+32+size]
+	default:
+		returnOutput = output[index : index+32]
+	}
```

Comment:
Shouldn't we check here if `len(output) >= offset+32`? And similarly that `output >= offset+32+size`? Otherwise a bad input and the entire node will crash with an out of bounds panic.